### ISSUE TITLE
keras chunk dataset shuffle in .fit

### ISSUE THREAD

**Issue by minsung-k** [keras-team/keras#18992]
Hi, I want to know how to shuffle train model with different dataset scenario in each epoch.

Assume I have a dataset and model like below.

```
import keras
from keras import layers
import numpy as np

(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()
x_train = np.reshape(x_train, (-1, 784))
x_test = np.reshape(x_test, (-1, 784))

x_train = x_train[:100]
x_valid = x_train[100:130]
x_test = x_test[:30]

y1 = [*range(30, 0, -1)]
y2 = [*range(40, 0, -1)]
y3 = [*range(10, 0, -1)]
y4 = [*range(20, 0, -1)]

y5 = [*range(13, 0, -1)]
y6 = [*range(17, 0, -1)]

y7 = [*range(12, 0, -1)]
y8 = [*range(18, 0, -1)]


# model
inputs = keras.Input(shape=(784,), name="digits")
x1 = layers.Dense(64, activation="relu")(inputs)
outputs = layers.Dense(1, name="predictions", activation = 'linear')(x2)
model = keras.Model(inputs=inputs, outputs=outputs)

# Instantiate an optimizer.
optimizer =  tf.keras.optimizers.SGD(learning_rate=1e-3)

# Compile a model.
model.compile(optimizer = "adam", loss = 'mse' )

# fit 
history  = model.fit(x_train, y_train, batch_size=1, epochs=10, verbose =2)


```

I only utilize 160 images from mnist and their are 8 scenarios, 4 scenarios are train dataset, and 2 are valid and test dataset each.

My question is is there a way to train a model with different train / valid / test dataset in each epoch.

For example,
epoch 1 train dataset ( [(x1,y1), (x2,y2) ,(x3,y3) , (x4,y4)], valid [ (x5,y5), (x6,y6)], test [(x7,y7) , (x8,y8) ] 

epoch 2  train dataset ( [(x3,y3), (x2,y2) ,(x7,y7) , (x6,y6)], valid [ (x5,y5), (x4,y4)], test [(x1,y1) , (x8,y8) ] 

I do not find a proper way to do it.

**Comment by sampathweb** [comment]
You could use `steps_per_epoch` if you want to train for only x number of batches in each epoch.  In general, its not appropriate to have a split like you have mentioned where `x4, y4` is in train for first epoch and in validation in second epoch.  You don't want to mix the train, valid and test datasets, otherwise the model may not perform as well with new data.

You can also get full control of what data is used for train in each epoch, by writing a custom training loop where you iterate over the different dataset batches and do the forward and backward pass on the model.

**Comment by minsung-k** [comment]
Appreciate it!
